server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -441,14 +443,24 @@ private static final class FailureCollector {
 
         List<FieldCapabilitiesFailure> build(Set<String> successfulIndices) {
             Map<Tuple<String, String>, FieldCapabilitiesFailure> indexFailures = new HashMap<>();
+            Set<String> exceptionsLogged = new HashSet<>();
             for (Map.Entry<String, Exception> failure : failuresByIndex.entrySet()) {
                 String index = failure.getKey();
                 Exception e = failure.getValue();
+                Throwable cause = ExceptionsHelper.unwrapCause(e);
+
+                // log 500 errors that are not Node/Shard unavailable type errors (and only once per Exception type)
+                if (ExceptionsHelper.status(e).getStatus() >= 500 && ExceptionsHelper.isNodeOrShardUnavailableTypeException(e) == false) {
+                    String loggingKey = e.getClass().getSimpleName() + (cause == null ? "" : cause.getClass().getSimpleName());
+                    if (exceptionsLogged.add(loggingKey)) {
```

Comment:
Just a though, if we dedup, it might be useful to also keep a counter for each key and add that to the exception message later. That would mean we need to delay logging until we have processed all failures, but I think that should be possible here.

Reply:
I also see we might have multiple indices with the same loggingKey, we would currently only log the name of the first one. Maybe sufficient, but maybe it makes sense to also collect a list here? I'm not really sure how useful that is yet so open for discussion.

Reply:
In my second code push, I've changed this section to use the existing dedup logic for logging as well. 

It has the downside that it requires an extra loop in the FailureCollector.build method, but the alternative is to build our own parallel dedup logic, which seems like the worse option?

The existing dedup logic is:
```
                    // we deduplicate exceptions on the underlying causes message and classname
                    // we unwrap the cause to e.g. group RemoteTransportExceptions coming from different nodes if the cause is the same
                    Throwable cause = ExceptionsHelper.unwrapCause(e);
                    Tuple<String, String> groupingKey = new Tuple<>(cause.getMessage(), cause.getClass().getName());
                    indexFailures.compute(
                        groupingKey,
                        (k, v) -> v == null ? new FieldCapabilitiesFailure(new String[] { index }, e) : v.addIndex(index)
                    );
```

This will achieve showing that the exception occurred on multiple indices (if it did), so this model gives both a grouping by Exceptions and an (implicit) count of how many indexes it occurred on.